### PR TITLE
MGMT-12870: Add missing dual-stack VIP CRUD for KubeAPI

### PR DESCRIPTION
**Pull request by mkowalski**
This PR implements a missing CRUD operations to handle dual-stack VIPs in case when KubeAPI is used.

Closes: [MGMT-12870](https://issues.redhat.com//browse/MGMT-12870)
Contributes-to: [MGMT-9915](https://issues.redhat.com//browse/MGMT-9915)
Implements: #4245

/cc @nmagnezi

### CODE REVIEW

File: internal/controller/controllers/common.go [hunk 1]
```diff
@@ -315,6 +315,30 @@ func machineNetworksEntriesToArray(entries []hiveext.MachineNetworkEntry) []*mod
 	}).([]*models.MachineNetwork)
 }
 
+func apiVipsArrayToStrings(vips []*models.APIVip) []string {
+	return funk.Map(vips, func(vip *models.APIVip) string {
```
Comment (omertuc) [inline internal/controller/controllers/common.go:319]: We have go 1.18 now we don't need funk anymore, we have https://github.com/samber/lo 

But you may keep using `funk` for consistency with the rest of this file if you want

**Comment by omertuc** [comment]
Could you give some short background about:

- What is "autocalculation" - calculation of what exactly, based on what exactly
- What does VIP have to do with SNO - SNO doesn't have a concept of VIPs, so what's up with all this VIP talk? What does it mean?

File: internal/controller/controllers/common.go (hunk 1 above)
Comment (mkowalski) [reply internal/controller/controllers/common.go:319]: Thanks for the suggestion! I had a look and it seems like instead of `funk.Map` I would be doing

```
import "github.com/samber/lo"

lo.Map[int64, string]([]int64{1, 2, 3, 4}, func(x int64, index int) string {
    return strconv.FormatInt(x, 10)
})
// []string{"1", "2", "3", "4"}
```

so I'd still need to import additional library and syntax-wise it's not much more clear than `funk.Map`. If there is no other push towards moving out of `funk`, I'd prefer to stay with it

Comment (omertuc) [reply internal/controller/controllers/common.go:319]: Actually I don't know why their README example is so ugly, if you use it properly it's simpler and gets rid of the cast at the end:

```go
func apiVipsArrayToStrings(vips []*models.APIVip) []string {
	return lo.Map(vips, func(vip *models.APIVip, _ int) string {
		return string(vip.IP)
	})
}
```

Not that performance matters here, but it's also much faster since it doesn't use ugly runtime `Interfaces{}` for everything

**Comment by mkowalski** [comment]
/hold

After discussing with Omer offline we have realized there is no any real need for users of SNO to define API VIPs explicitly in order to consume this feature. Therefore I will remove the part of the code that allows doing so.

File: internal/cluster/validations/validations.go [hunk 2]
```diff
@@ -442,6 +442,7 @@ func ValidateClusterCreateIPAddresses(ipV6Supported bool, clusterId strfmt.UUID,
 	targetConfiguration.IngressVips = params.IngressVips
 	targetConfiguration.UserManagedNetworking = params.UserManagedNetworking
 	targetConfiguration.VipDhcpAllocation = params.VipDhcpAllocation
+	targetConfiguration.HighAvailabilityMode = params.HighAvailabilityMode
```
Comment (mkowalski) [inline internal/cluster/validations/validations.go:445]: todo myself: extract this part to separate commit

File: subsystem/kubeapi_test.go [hunk 3]
```diff
@@ -3917,12 +3917,24 @@ var _ = Describe("[kube-api]cluster installation", func() {
 			return aci.Status.APIVIP
 		}, "30s", "1s").Should(Equal(aciSpec.APIVIP))
 
+		By("Ensure APIVIPs exist in status")
+		Eventually(func() int {
+			aci := getAgentClusterInstallCRD(ctx, kubeClient, installkey)
+			return len(aci.Status.APIVIPs)
+		}, "30s", "1s").ShouldNot(Equal(0))
```
Comment (mkowalski) [inline subsystem/kubeapi_test.go:3924]: todo myself: add a test case to check apivips[0] for a value of the IP address

**Review by nmagnezi — APPROVED** [review]
/lgtm
